src/libsystemd-network/sd-dhcp-lease.c

```diff
@@ -1136,8 +1156,8 @@ int dhcp_lease_load(sd_dhcp_lease **ret, const char *lease_file) {
                 *vendor_specific_hex = NULL,
                 *lifetime = NULL,
                 *t1 = NULL,
-                *t2 = NULL,
-                *options[SD_DHCP_OPTION_PRIVATE_LAST - SD_DHCP_OPTION_PRIVATE_BASE + 1] = {};
```

Comment:
Thanks! I looked at that code and couldn't figure out where the memory leak came from because I assumed some `_cleanup_` function took care of `options` :-) Turns out only the first option was taken care of apparently.

FWIW I opened https://github.com/systemd/systemd/pull/22311 where I added `fuzz-lease-load` along with the file triggering the memory leak. I think it would be easier to make sure all the issues are gone if it was backported to this PR.